lib/node_modules/@stdlib/lapack/base/dlaswp/benchmark/benchmark.js

```diff
@@ -0,0 +1,105 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var uniform = require( '@stdlib/random/array/uniform' );
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var pkg = require( './../package.json' ).name;
+var dlaswp = require( './../lib/dlaswp.js' );
+
+
+// FUNCTIONS //
+
+/**
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} N - number of elements along each dimension
+* @returns {Function} benchmark function
+*/
+function createBenchmark( N ) {
+	var IPIV;
+	var A;
+
+	IPIV = discreteUniform( 5, 0, N-1, {
+		'dtype': 'int32'
+	});
+	A = uniform( N*N, -10.0, 10.0, {
+		'dtype': 'float64'
+	});
+	return benchmark;
+
+	/**
+	* Benchmark function.
+	*
+	* @private
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var z;
+		var i;
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			z = dlaswp( 'column-major', N, A, N, 0, 2, IPIV, 1 );
```

Comment:
If I put any number greater than 2 here, it throws NaN somehow.

Reply:
Odd. I just ran the benchmarks locally and they run fine.

Reply:
Which argument are you referring to?

Reply:
For `k2`?

Reply:
For `k2`. Okay. I'm able to reproduce.

Reply:
Ah, well, yes, of course. It cannot be `5` for all `N`. We provide some matrices which are `3x3` and `5` exceeds the number of rows.

Reply:
I think your concern is that IPIV has 5 elements, but we are only interchanging 3 rows. That's fine, but also, we can just adjust IPIV to only contain 3 elements.

Reply:
I've updated the benchmarks accordingly. We now also measure performance across the number of interchanged rows.